src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2345,13 +2345,13 @@ class MethodArityHistogram {
     const int N = MIN2(9, n);
     double sum = 0;
     double weighted_sum = 0;
-    for (int i = 0; i <= n; i++) { sum += histo[i]; weighted_sum += i*histo[i]; }
+    for (int i = 0; i <= n; i++) { sum += (double)histo[i]; weighted_sum += (double)(i*histo[i]); }
```

Comment:
So hist is a uint64_t*, why can't we make sum and weighted_sum just uint64_t?

Reply:
I think there were less casts this way, since sum and weighted sum had to be turned into double more than histo[i], but it does make more sense to use int for summing the numbers, and cast to double for printing.

Reply:
I don't know how big these values are, but if they were large then uint64_t sum could overflow.

Reply:
uint64_t count = (uint64_t)method->compiled_invocation_count();

They seem to be a sum of compiled_invocation_count() so maybe double is best.